src/utils/deployPeertube.ts

```diff
@@ -42,156 +39,53 @@ export default async function deployPeertube(data: VM, profile: IProfile) {
 
   // Make sure the name is valid
   name = await getSuitableGateway(client, name);
-  // console.log(name);
 
-  // Gateway node Id and domain
-  const gwNodeId = 8;
-  const nodes = new Nodes(GridClient.config.graphqlURL, GridClient.config.rmbClient["proxyURL"]); // prettier-ignore
-  const gwDomain = await getNodeDomain(nodes, gwNodeId);
-  domain = `${name}.${gwDomain}`;
-  // console.log(gateway);
+  // Dynamically select node to deploy the gateway
+  const nodes = new Nodes(
+    GridClient.config.graphqlURL,
+    GridClient.config.rmbClient["proxyURL"]
+  );
 
-  // // define a network
-  const network = new NetworkModel();
-  network.name = name + "NW";
-  network.ip_range = "10.1.0.0/16";
+  const filter = new FilterOptions();
+  filter.gateway = true;
 
-  // // deploy redis
-  await deployRedis(client, network, nodeId, name);
+  const selectedNode = randomChoice(await nodes.filterNodes(filter));
```

Comment:
what happens if there're none?

Reply:
the deployment fails with this error msg
![Screenshot from 2021-12-15 14-18-42](https://user-images.githubusercontent.com/32527347/146185441-14c1a0f4-2820-413f-9e8d-63b8166455a8.png)

Reply:
it's ok for now. cool.